Template based type in OBJECTS clause not working

Template based type in OBJECTS clause not working. 

The bellow code spinet is not parsing:
```
ABSTRACT SPEC java.lang.String<T>
OBJECTS:
	<T> foo;
```
Error: 
`java.lang.AssertionError: Unexpected errors: XtextSyntaxDiagnostic: null:3 required (...)+ loop did not match anything at input '<'`

--- a/br.unb.cic.mcsl.tests/src/br/unb/cic/mcsl/tests/MetaCrySLParsingTest.xtend
+++ b/br.unb.cic.mcsl.tests/src/br/unb/cic/mcsl/tests/MetaCrySLParsingTest.xtend
@@ -32,6 +32,19 @@ class MetaCrySLParsingTest extends MetaCrySLAbstractTests {
 		val errors = result.eResource.errors
 		Assert.assertTrue('''Unexpected errors: «errors.join(", ")»''', errors.isEmpty)
 	}
+
+	@Test
+	def void loadBasicModelWithTypeObject() {
+		val file = super.readFileIntoString(BR_UNB_CIC_METACRYSL_TESTS_FILES + "basicModelWithTypeObject.cryptsl")
+		val result = super.parseHelper.parse(file)
+		Assert.assertNotNull(result)
+		val errors = result.eResource.errors
+		Assert.assertTrue('''Unexpected errors: «errors.join(", ")»''', errors.isEmpty)
+		
+		val object = result.metaCrySL.objectSpec.getObjects().get(0)
+		Assert.assertEquals(object.objectName, "foo")
+		Assert.assertEquals(object.objectTypeParam, "T")
+	}
 	
 	@Test
 	def void loadBasicModel() {

--- a/br.unb.cic.mcsl.tests/src/br/unb/cic/mcsl/tests/files/basicModelWithTypeObject.cryptsl
+++ b/br.unb.cic.mcsl.tests/src/br/unb/cic/mcsl/tests/files/basicModelWithTypeObject.cryptsl
@@ -1,0 +1,8 @@
+ABSTRACT SPEC java.lang.String<T>
+OBJECTS:
+	<T> foo;
+EVENTS
+	c1 : method1();
+	c2 : method2(int, int);
+	c3 : method2(_);
+	cs := c1 | c2 ; 

--- a/br.unb.cic.mcsl/src/br/unb/cic/mcsl/MetaCrySL.xtext
+++ b/br.unb.cic.mcsl/src/br/unb/cic/mcsl/MetaCrySL.xtext
@@ -40,7 +40,7 @@ ObjectSpec: {ObjectSpec} 'OBJECTS' ':' objects += Object+ ;
  * variable that can be used in the remaining 
  * sections of the specification. 
  */
-Object: {Object} type = JvmTypeReference varName = ID ';' ;
+Object: {Object} ((ObjectType = JvmTypeReference) | ("<"ObjectTypeParam = ValidID">")) ObjectName = ID ';' ;
 
 /* ForbiddenSpec */
 ForbiddenSpec: {ForbiddenSpec} 'FORBIDDEN' forbidenMethods += ForbiddenMethod+;

